pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,203 @@
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_proxyconfig")
+
+const (
+	singletonName          = "instance"
+	openshiftNS            = "openshift-logging"
+	proxyName              = "cluster"
+	trustBundleConfigmapNS = "openshift-config"
+)
+
+// Add creates a new ClusterLogging Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	if err := configv1.Install(mgr.GetScheme()); err != nil {
+		return &ReconcileProxyConfig{}
+	}
+
+	return &ReconcileProxyConfig{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("proxyconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// We only care about a configmap source with a specific name/namespace,
+	// so filter events before they are provided to the controller event handlers.
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return handleConfigMap(e.MetaNew)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	if err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred); err != nil {
+		return err
+	}
+
+	// Watch for changes to the proxy resource.
+	if err = c.Watch(&source.Kind{Type: &configv1.Proxy{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// handleConfigMap returns true if meta namespace is "openshift-config".
+// "openshift-config" is the namespace for one or more
+// ConfigMaps that contain user provided trusted CA bundles.
+func handleConfigMap(meta metav1.Object) bool {
+	return meta.GetNamespace() == "openshift-config"
+}
+
+var _ reconcile.Reconciler = &ReconcileProxyConfig{}
+
+// ReconcileProxyConfig reconciles a ClusterLogging object
+type ReconcileProxyConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func proxyNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Name: "cluster",
+	}
+}
+
+// Reconcile reads that state of the cluster for a cluslter-scoped
+// named "cluster" or a configmap object in namespace "openshift-config".
```

Comment:
you shouldn't be watching the configmap in openshift-config, you should have your own configmap that's labeled for injection which you watch.  why are you watching this?

Reply:
Ok.  That was one of my confusions.  (and what Daneyon commented on the code, which I did not understand well back then...)  I'm fixing it.  Thanks!

> @nhosoi you should not model your implementation after the network-operator. Operators should use CVO for proxy env var injection and create a configmap for trusted ca injection. Additional details can be found in the proxy design doc. Here is an example implementation of proxy env's and this one for trusted ca.

Reply:
does this still need to be addressed? can we update this comment to be more reflective of what we're doing?